src/fdcache.cpp

```diff
@@ -908,13 +908,46 @@ bool PageList::Serialize(CacheFileStat& file, bool is_output)
     // loaded
     Clear();
 
-    // load(size)
+    // load head line(for size and inode)
+    off_t total;
+    ino_t cache_inode;                  // if this value is 0, it means old format.
     if(!getline(ssall, oneline, '\n')){
       S3FS_PRN_ERR("failed to parse stats.");
       delete[] ptmp;
       return false;
+    }else{
+      istringstream sshead(oneline);
+      string        strhead1;
+      string        strhead2;
+
+      // get first part in head line.
+      if(!getline(sshead, strhead1, ':')){
+        S3FS_PRN_ERR("failed to parse stats.");
+        delete[] ptmp;
+        return false;
+      }
+      // get second part in head line.
+      if(!getline(sshead, strhead2, ':')){
+        // old head format is "<size>\n"
+        total       = wrap_strtoofft(strhead1.c_str());
+        cache_inode = 0;
+      }else{
+        // current head format is "<inode>:<size>\n"
+        total       = wrap_strtoofft(strhead1.c_str());
+        cache_inode = static_cast<ino_t>(wrap_strtoofft(strhead2.c_str()));
+        if(0 == cache_inode){
+          S3FS_PRN_ERR("wrong inode number in parsed cache stats.");
+          delete[] ptmp;
+          return false;
+        }
+      }
+    }
+    // check inode number
+    if(0 != cache_inode && cache_inode != inode){
+      S3FS_PRN_ERR("differ inode and inode number in parsed cache stats.");
```

Comment:
I'm trying to understand what the strictness helps with -- do you worry about an external process modifying the cache file while s3fs is running?

Reply:
Although Including editing by an external process, I am concerned about problems(cache inconsistency) due to other factors, too.

The cache file is opened by s3fs when the file(object) is opened.
At that time, load the stats file of the cache file and s3fs knows the contents of the cache file(which area is loaded).
This may not be good, but s3fs uses two files to manage one piece of information.
Thus I believe these need to be consistent.
Therefore, the inode number was added to cache stats file for that it pointed to correct cache file.
If this inode number is not as same as the cache file's, s3fs should not load the cache file and its stats.